mmrazor/datasets/utils.py

```diff
@@ -1,15 +1,63 @@
+import os
+import random
+
 from torch.utils.data import random_split
 
 
-def split_dataset(dataset):
+def split_dataset_txt(txt_path, save_dir):
+    index_pool = dict()
+    with open(txt_path, 'r') as f:
+        for line in f:
+            line = line.strip()
+            (filename, index) = line.split(' ')
+            if index not in index_pool:
+                index_pool[index] = []
+            index_pool[index].append(filename)
+
+    assert len(index_pool) == 1000
+
+    train_txt = list()
+    val_txt = list()
+    for index, filenames in index_pool.items():
+        filenames_val = random.sample(filenames, 50)
+        for name in filenames_val:
+            filenames.remove(name)
+        filenames_train = filenames
+        for name in filenames_train:
+            train_txt.append(f'{name} {index}')
+        for name in filenames_val:
+            val_txt.append(f'{name} {index}')
+
+    assert len(val_txt) == 50000
+
+    with open(os.path.join(save_dir, 'greedynas_train.txt'), 'w') as f:
```

Comment:
`greedynas_train.txt` should be `train/val_part_1.txt` according to the  origin txt name